### ISSUE TITLE
Video probleme (compilation not failed)

### ISSUE THREAD

**Issue by tuxme** [SteveMaddison/cabrio#14]
Hello everyone, I just started my project Bartop arcade DIY and I chose cabrio-fe because he was the best on linux and I can not suporter the mouse. I have a brief porbleme when compiling on video (see below) and the video never read. I'm running debian squeeze amd 64. Help me.

Oh and here is the link to my youtube account and see the DEADPOOL CAB ARCADE CABINET :)

http://www.youtube.com/feed/UCdyVxCbkSDXHKL_B7xPYwag/u

See you

berzerking@deadpoolcab:~/cabrio$ make
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o main.o -c main.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o ogl.o -c ogl.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o sdl_wrapper.o -c sdl_wrapper.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o config.o -c config.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o bg.o -c bg.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o menu.o -c menu.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o game_sel.o -c game_sel.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o game.o -c game.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o font.o -c font.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o hint.o -c hint.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o platform.o -c platform.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o submenu.o -c submenu.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o sound.o -c sound.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o event.o -c event.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o key.o -c key.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o control.o -c control.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o setup.o -c setup.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o sdl_ogl.o -c sdl_ogl.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o video.o -c video.c
video.c: In function ‘video_init’:
video.c:76: warning: ‘url_set_interrupt_cb’ is deprecated (declared at /usr/include/libavformat/avio.h:201)
video.c: In function ‘video_open’:
video.c:308: warning: ‘av_open_input_file’ is deprecated (declared at /usr/include/libavformat/avformat.h:1170)
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o packet.o -c packet.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o frame.o -c frame.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o category.o -c category.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o focus.o -c focus.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o emulator.o -c emulator.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o snap.o -c snap.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o media.o -c media.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o location.o -c location.c
gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o lookup.o -c lookup.c
gcc -o cabrio main.o ogl.o sdl_wrapper.o config.o bg.o menu.o game_sel.o game.o font.o hint.o platform.o submenu.o sound.o event.o key.o control.o setup.o sdl_ogl.o video.o packet.o frame.o category.o focus.o emulator.o snap.o media.o location.o lookup.o -lGL -lSDL -lSDL_image -lSDL_gfx -lSDL_ttf -lSDL_mixer -lGLU -lxml2 -lavutil -lavformat -lavcodec -lswscale

**Comment by tallerthenyou** [comment]
Does it seem like it might be related to issue #13?

**Comment by tuxme** [comment]
yes, i put the log of the compilation.  i suposed the probleme is in the compilation : gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o video.o -c video.c
video.c: In function ‘video_init’:
video.c:76: warning: ‘url_set_interrupt_cb’ is deprecated (declared at /usr/include/libavformat/avio.h:201)
video.c: In function ‘video_open’:
video.c:308: warning: ‘av_open_input_file’ is deprecated (declared at /usr/include/libavformat/avformat.h:1170)

**Comment by tuxme** [comment]
compile works ... 

modify  url_set_interrupt_cb( video_stopped );
in      avio_set_interrupt_cb( video_stopped );
    and
modify  if( av_open_input_file( &format_context, filename, NULL, 0, NULL ) != 0 ) {
in      if( avformat_open_input( &format_context, filename, NULL, NULL ) != 0 ) {

dont know if video works i try by ssh (not front my cab;)) say you in 10 hours ;)

**Comment by tuxme** [comment]
Hello ...
so ... that don't work ... erf

See you :)

**Comment by tuxme** [comment]
ok, i have video and sound now ;) 

here is my patch :)

http://tuxme.net/data/cabrio/video.patch

i commit this modification in my repo : https://github.com/tuxme/tuxbrio

see you ;)

PS:  for information a try it on an Ubuntu 13.04 ... say you if it's work on debian squeeze soon :)

**Comment by tuxme** [comment]
arf ... don't work with debian squeeze ... : 

gcc -g -Wall -DDATA_DIR=\"/usr/share/cabrio\" -I./include -I/usr/include/libxml2 -o video.o -c video.c
video.c: In function âvideo_openâ:
video.c:309: warning: âav_open_input_fileâ is deprecated (declared at /usr/include/libavformat/avformat.h:1170)
video.c:409: error: expected â=â, â,â, â;â, âasmâ or â**attribute**â before âint_cbâ
video.c:409: error: âint_cbâ undeclared (first use in this function)
video.c:409: error: (Each undeclared identifier is reported only once
video.c:409: error: for each function it appears in.)
video.c:409: error: expected expression before â{â token
video.c:410: error: âAVFormatContextâ has no member named âinterrupt_callbackâ
make: **\* [video.o] Erreur 1

For information if i remove the 2 last lines
(static const AVIOInterruptCB int_cb={ video_stopped, NULL};
format_context->interrupt_callback=int_cb;)
first video launch but when you change game ... can"t open :)
